### PR TITLE
[ShapeOpt] avoid division by 0 when normalizing surface normals

### PR DESCRIPTION
**Pull request by armingeiser**

### CODE REVIEW

File: applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h [hunk 1]
```diff
@@ -409,7 +409,12 @@ class GeometryUtilities
         {
             const array_1d<double,3>& area_normal = node_i.FastGetSolutionStepValue(NORMAL);
             array_3d& normalized_normal = node_i.FastGetSolutionStepValue(NORMALIZED_SURFACE_NORMAL);
-            noalias(normalized_normal) = area_normal/norm_2(area_normal);
+
+            const double norm2 = norm_2(area_normal);
+            KRATOS_ERROR_IF(norm2<1e-10) << "Norm2 of normal for node "
```
Comment (dbaumgaertner) [inline applications/ShapeOptimizationApplication/custom_utilities/geometry_utilities.h]:
```suggestion
            KRATOS_ERROR_IF(norm2<1e-10) << "CalculateUnitNormals: Norm2 of normal for node "
```

**Review by dbaumgaertner — DISMISSED** [review]
Looks good. Just a minor suggestion, but your call. Just wondering, in which case it happend, that the nodal area was zero?

**Comment by armingeiser** [comment]
It happens if there is a node in the design surface that is not connected to any surface condition. This is allowed if no projection (or area weighting) is used, but gives NAN when projecting on the normals...

**Comment by armingeiser** [comment]
@dbaumgaertner applied your suggestion - can you reapprove?

**Review by dbaumgaertner — APPROVED** [review]

**Pull request merged**